docs/_source/docs/faq.rst

```diff
@@ -83,6 +83,71 @@ used.
+In order to call AWS services in your custom hook/resolver/template handler properly, you should use
+the IAM configurations of the stack where the resolver is being used (unless you need to use a
+different configuration for a specific reason). This means your hook/resolver/handler should honor the
+``profile``, ``region``, and ``iam_role`` configurations as set for your project and/or Stack Config.
+Simply invoking ``boto3.client('s3')`` is _not_ going to regard those and could end up using the
+wrong credentials or not even working.
+
+There is a simple interface available for doing this properly, the
+:py:class:`sceptre.connection_manager.ConnectionManager`. The ConnectionManager is an interface that
+will be pre-configured with each stack's profile, region, and iam_role and will be ready for you to use.
+If you are using an `iam_role`, it will automatically assume that role via STS for making calls to
+AWS so you can just use it the way you want. It is accessible on hooks and resolvers via
+``self.stack.connection_manager`` and on template_handlers via ``self.connection_manager``.
+
+There are three public methods on the ConnectionManager:
+
+- :py:meth:`sceptre.connection_manager.ConnectionManager.call` can be used to directly call a boto3
+  API method on any api service and return the result from boto3. This is perfect when you just need
+  to invoke a boto3 client method without any additional capabilities.
+- :py:meth:`sceptre.connection_manager.ConnectionManager.get_session` can be used to get a boto3 Session
+  object. This is very useful if you need to work with Boto3 Resource objects (like an s3 Bucket) or
+  if you need to create and pass the bot3 session, client, or resource to a third-party framework.
+- :py:meth:`sceptre.connection_manager.ConnectionManager.create_session_environment_variables` creates
+  a dictionary of environment variables used by AWS sdks with all the relevant connection information.
+  This is extremely useful if you are needing to invoke other SDKs using ``subprocess`` and still need
+  the Stack's connection information honored.
+
+
+Using the connection manager, you can use `boto3 <https://boto3.amazonaws.com/v1/documentation/api/latest/index.html>`_
+to perform any AWS actions you need:
+
+.. code-block:: python
+
+   # For example, in your custom resolver:
+   def resolve(self):
+       # You can invoke a lower-level service method like...
+       obj = self.stack.connection_manager.call('s3', 'get_object', {'Bucket': 'my-bucket', 'Key': 'my-key'})
+       # Or you can create higher-level resource objects like...
+       bucket = self.stack.connection_manager.get_session().resource('s3').Bucket('my-bucket')
+       # Or if you need to invoke a third-party tool via a subprocess, you can create the necessary environment
+       # variables like this:
+       environment_variables = self.stack.connection_manager.create_session_environment_variables(
+           include_system_envs=True
```

Comment:
That's a question I kept mulling over myself. In all of _my_ usecases, I wanted the system envs. However, that's somewhat a mixing of concerns (to an extent). It could be someone would want just the dict of envs they care about and then they can do whatever else they want with it.

The issue is that it's not as simple as just adding in the system envs if you want them mixed in. There are certain alterations that are necessary too... So maybe I always include the system envs? The problem is that then you get a super full list of envs if you don't want them all.

Reply:
I'm also wondering if we should update the `!cmd` hook and `!rcmd` resolvers to use the envs from the stack's connection manager to do what they do. This has the downside that it _could_ be breaking for certain stack configurations. However, it has the upside that it would allow those hook and resolver subprocesses to actually work with the stack's iam configurations without needing any sort of special environment magic.

Reply:
> The problem is that then you get a super full list of envs if you don't want them all.

is there a downside to having the full list?  Wouldn't it be better to just let the user decide how they want to manage that list after getting it?

> I'm also wondering if we should update the !cmd hook and !rcmd resolvers to use the envs from the stack's connection manager to do what they do

I was thinking this myself, there's a [related issue](https://github.com/Sceptre/sceptre-ssm-resolver/issues/13) in the SSM resolver repo.   I'm fine with updating on an as needed basis but by all means don't let me stop you if you are really motivated to update those sceptre plugins.  Since it looks like an issue came in for the ssm resolver maybe that might be one that can be updated first?  it would be nice to have at least one as a best practice example.

Reply:
What do you think about updating `!cmd` and `!rcmd`, considering those changes MIGHT be breaking for some users? We could document that this is a breaking change and include this as a part of a major version release.

And yeah, once we get this change deployed, I'll go and update the SAM and CDK resolvers to simplify their code and use this new, better interface.

Reply:
As far as the SSM resolver issue goes, it's a little different of an issue. We could fix that issue now, probably, without waiting on this PR to get deployed.

Reply:
> What do you think about updating !cmd and !rcmd, considering those changes MIGHT be breaking for some users? We could document that this is a breaking change and include this as a part of a major version release.

I'm fine with it.  Users do have the option to downgrade to an older version of a plugin if they like.

Reply:
Yeah, but `!cmd` and `!rcmd` is packaged _with_ Sceptre.

Reply:
Actually, I take that back. `!cmd` is packaged with sceptre. `!rcmd` is not; it's a dependency of Sceptre.

Reply:
yep, users can downgrade the entire sceptre version or downgrade just the specific plugin version.